### DISCUSSION TITLE
How to position measure number?

### DISCUSSION THREAD

**Discussion by ego-lay-atman-bay** [0xfe/vexflow#1493] [Q&A]
How do I position the measure numbers? I don't want the numbers to be directly above the barline.

**Comment by rvilarl** [comment]
It seems that there is no way to do that now. Where would you like to position them? Can you share an example?

**Comment by ego-lay-atman-bay** [comment]
I was trying to get it to move a little to the right, because I'm making each measure a separate element, and the number was getting cut off. I managed to find a hacky solution.
```
output.querySelector('text').setAttribute('x', 0)
```
(The default x position is -2)
This is called after the stave is rendered.

I do also have another question, why does the measure number not have a class?

**Reply by rvilarl** [reply]
> I do also have another question, why does the measure number not have a class?

You are right, I believe, this does not seem to be right.

**Reply by sschmidTU** [reply]
In OSMD we just take the measure's position, add an offset if needed, and use `renderText` in the vexflow backend.
<img width="594" alt="image" src="https://user-images.githubusercontent.com/33069673/208392048-1c326526-a41e-48c5-8941-678952777cbb.png">

(I've added an offset of 10 pixels here, though in OSMD we don't do that, because it can overlap other elements or at least lead to being positioned higher up by our collision system, as you see in measure 5)
Without the offset:
<img width="593" alt="image" src="https://user-images.githubusercontent.com/33069673/208392381-8f4e81cb-9dbc-4842-95e0-c556cf6dedd4.png">

Maybe the measure number was cut off because you didn't leave enough margin in the canvas size for it?

**Comment by ego-lay-atman-bay** [comment]
What I'm doing is having each measure have a different canvas (button) (because I need the user to be able to select a certain measure), so I guess it wasn't giving the measure number enough room.

**Comment by ego-lay-atman-bay** [comment]
Oh yeah, the specific version I'm using is 3.6.0

edit: oh, I didn't realize that 4.1.0 was the most recent version. I just got it from a link in the install guide.

**Comment by ego-lay-atman-bay** [comment]
actually, after finally being able to revisit this, I found out that I am indeed using version 4.1.0. I don't know where I got the idea that I was using 3.6.0, but at least I don't have to update it.
